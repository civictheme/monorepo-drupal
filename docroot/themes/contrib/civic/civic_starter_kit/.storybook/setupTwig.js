const twigDrupal = require('twig-drupal-filters');

/**
 * Configures and extends a standard twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
