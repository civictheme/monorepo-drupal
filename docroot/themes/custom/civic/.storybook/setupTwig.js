const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');

module.exports.namespaces = {
  atoms: resolve(__dirname, '../', 'components/01-atoms'),
  molecules: resolve(__dirname, '../', 'components/02-molecules'),
  organisms: resolve(__dirname, '../', 'components/03-organisms'),
  templates: resolve(__dirname, '../', 'components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
