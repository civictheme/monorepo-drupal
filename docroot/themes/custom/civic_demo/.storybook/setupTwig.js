const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');


module.exports.namespaces = {
  atoms: resolve(__dirname, '../', '.storybook-components/01-atoms'),
  molecules: resolve(__dirname, '../', '.storybook-components/02-molecules'),
  organisms: resolve(__dirname, '../', '.storybook-components/03-organisms'),
  templates: resolve(__dirname, '../', '.storybook-components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
