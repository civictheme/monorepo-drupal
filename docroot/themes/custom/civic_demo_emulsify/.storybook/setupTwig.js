const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');
const twigBEM = require('bem-twig-extension');
const twigAddAttributes = require('add-attributes-twig-extension');

module.exports.namespaces = {
  atoms: [
    resolve(__dirname, '../../civic_emulsify', 'components/01-atoms'),
    resolve(__dirname, '../', 'components/01-atoms'),
  ],
  molecules: [
    resolve(__dirname, '../../civic_emulsify', 'components/02-molecules'),
    resolve(__dirname, '../', 'components/02-molecules'),
  ],
  organisms: [
    resolve(__dirname, '../../civic_emulsify', 'components/03-organisms'),
    resolve(__dirname, '../', 'components/03-organisms'),
  ],
  templates: [
    resolve(__dirname, '../../civic_emulsify', 'components/04-templates'),
    resolve(__dirname, '../', 'components/04-templates'),
  ]
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigBEM(twig);
  twigDrupal(twig);
  twigAddAttributes(twig);
  return twig;
};
