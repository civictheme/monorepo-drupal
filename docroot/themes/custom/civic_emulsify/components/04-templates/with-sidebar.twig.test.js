import { join } from 'path';
import { render, Twig } from 'twig-testing-library';

import { setupTwig, namespaces } from '../../.storybook/setupTwig';

setupTwig(Twig);

describe('with-sidebar', () => {
  it('can render a template with a sidebar', async () => {
    const { container } = await render(
      join(__dirname, 'with-sidebar.twig'),
      {},
      namespaces,
    );

    expect(container).toMatchSnapshot();
  });
});
