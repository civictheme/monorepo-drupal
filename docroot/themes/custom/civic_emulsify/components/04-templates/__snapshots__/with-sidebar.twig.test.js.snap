// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with-sidebar can render a template with a sidebar 1`] = `
<div>
  


    

  <div
    class="layout-container"
  >
    
      


    <header
      class="header"
    >
      
  
      <div
        class="header__inner"
      >
        
    
        <div
          class="header__primary"
        >
          
      
          <div
            class="header__branding"
          >
            
                  


            <a
              class="logo-link"
              href="/"
            >
              
      


  

              <img
                alt="Logo"
                class="logo__image"
                src="logo.png"
              />
              
  
            </a>
            
              
          </div>
          
      
          <div
            class="header__menu"
          >
            
                  


            <nav>
              
  
              <a
                class="toggle-expand"
                href="#"
                id="toggle-expand"
              >
                
    
                <span
                  class="toggle-expand__open"
                >
                  
      






                  <svg
                    class="toggle-expand__icon"
                  >
                    
      
                    <use
                      xlink:href="icons.svg#menu"
                    />
                    

                  </svg>
                  
      
                  <span
                    class="toggle-expand__text"
                  >
                    Main Menu
                  </span>
                  
    
                </span>
                
    
                <span
                  class="toggle-expand__close"
                >
                  
      
                  <span
                    class="toggle-expand__text"
                  >
                    Close
                  </span>
                  
    
                </span>
                
  
              </a>
              
  
              <div
                class="main-nav"
                id="main-nav"
              >
                
    




    

              </div>
              

            </nav>
            
              
          </div>
          
    
        </div>
        
    
        <div
          class="header__secondary"
        >
          
              

          
        </div>
        
  
      </div>
      

    </header>
    
        
    <div
      class="main main--with-sidebar"
    >
      
    
      <a
        id="main-content"
        tabindex="-1"
      />
      
          
      <aside
        class="main-sidebar"
        role="complementary"
      >
        
            

        <section
          class="place-holder"
        >
          
  
          <div
            class="place-holder__content"
          >
            
    Secondary Content
  
          </div>
          

        </section>
        
        
      </aside>
      
        
      <main
        class="main-content main-content--with-sidebar"
        role="main"
      >
        
          

        <section
          class="place-holder"
        >
          
  
          <div
            class="place-holder__content"
          >
            
    Primary Content
  
          </div>
          

        </section>
        
      
      </main>
      
  
    </div>
    
      

    <footer
      class="footer"
    >
      
  
      <div
        class="footer__inner"
      >
        
    
        <div
          class="footer__social"
        >
          
              


          <ul
            class="social-menu"
          >
            
        
          </ul>
          
          
        </div>
        
    
        <div
          class="footer__menu"
        >
          
              




    
          
        </div>
        
  
      </div>
      

    </footer>
    
  
  </div>
  

</div>
`;
